### PR TITLE
Scripts/BlackrockDepths: Small correction for final boss room

### PR DESCRIPTION
**Pull request by Jinnaix**
**Changes proposed:**

-  Removing call for help on Boss
- Adding creature formation to all units instead 

Call for help only targets units visible for the boss - this leaves out several units in the final room.

**Target branch(es):**

- [X] 3.3.5

**Issues addressed:**

None

**Tests performed:**

Builds and tested

**Known issues and TODO list:** (add/remove lines as needed)

Irrelevant in this case but just a reminder: https://github.com/TrinityCore/TrinityCore/issues/17563

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_emperor_dagran_thaurissan.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ class boss_emperor_dagran_thaurissan : public CreatureScript
             void JustEngagedWith(Unit* /*who*/) override
             {
                 Talk(SAY_AGGRO);
-                me->CallForHelp(VISIBLE_RANGE);
+                // me->CallForHelp(VISIBLE_RANGE); -- Only calls in line of sight, leaves out the Units behind the pillars
```
Comment (Aokromes) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_emperor_dagran_thaurissan.cpp:67]: delete thee row, no comment it.

**Comment by Jildor** [comment]
maybe better modify CallForHelp method and add bool ignoreLoS after float radius?

**Comment by offl** [comment]
This way we will be not able to create formations for packs(primary key will not allow it)
Instead we need to create another param for CallForHelp to allow it to select creatures not in LoS

Or update formations
There are cases https://github.com/TrinityCore/TrinityCore/issues/24786#issuecomment-643774655 which cannot be done with CallForHelp and only with formations

**Comment by Jinnaix** [comment]
> This way we will be not able to create formations for packs(primary key will not allow it)
> Instead we need to create another param for CallForHelp to allow it to select creatures not in LoS
> 
> And update formations
> There are cases [#24786 (comment)](https://github.com/TrinityCore/TrinityCore/issues/24786#issuecomment-643774655) which cannot be done with CallForHelp and only with formations
> 
> In this particular case better to use CallForHelp

I did this change some months ago on my private fork and im sure i checked if the unit packs are standing close enough to each other to correctly be pulled as a pack on aggro.

**Comment by offl** [comment]
Point still stands, we will be not able to use formations or CallForHelp in other cases
And I'm able to pull less than I should

**Comment by Jinnaix** [comment]
Im fine with this getting closed. I personally see the need for sub-formations in this and other cases.
Especially in tbc dungeons there was a lot of situations where previous bosses get pulled if you sneak past them and aggro the final boss, while the previous bosses also had a room pull mechanic.

**Comment by offl** [comment]
We barely can use both options currently
If someone will want to make formations, he will need to remove your formations and restore CallForHelp
In case of Pandemonius I can't use neither CallForHelp or formations

**Comment by jackpoz** [comment]
Should a new issue be created about the "ignoreLoS" parameter ?

**Comment by Jildor** [comment]
That is what I think would be more useful, but I don't know if it is what offl needs in the case of pandemonius or if need something else, because if it is only for this case (thaurissan) I don't see it necessary

**Comment by offl** [comment]
I remember https://youtu.be/UcOi3-Hj-Y4?t=385 but CallForHelp will also pull Whitemane out of door and that's not good

**Comment by offl** [comment]
> Instead we need to create another param for CallForHelp to allow it to select creatures not in LoS

Nah
It's a temp solution, we don't really need it
So yeah, it will be used in few cases

**Comment by jackpoz** [comment]
> It's a temp solution, we don't really need it

@offl what is the long term solution ? I'm just trying to figure out if you need someone to fix some C++ issues or if you have everything in place and the fix is going to be a SQL-only update

**Comment by Jildor** [comment]
in case you need to add the option to ignore LoS in CallForHelp, let me know, I can do a PR

**Comment by Jinnaix** [comment]
Just a reminder that as of right now, the boss room behavior in BRD is not correct.
But i think generating aggro with CallForHelp would trigger a creature formation, correct? Maybe in that case we group the trash packs in that room.

**Comment by offl** [comment]
> what is the long term solution ?

I wish I knew

1. We indeed can use CallForHelp in some cases, it's fine
2. CallForHelp ignoreLoS probably will be used few times, is it worth it?
3. we currently can't use formations if not by pathing creatures because they are broken https://github.com/TrinityCore/TrinityCore/issues/17563#issuecomment-753526821 https://github.com/TrinityCore/TrinityCore/issues/25494
4. even if 3 will be fixed right now I don't know how to use formations to link packs of trash and at the same time link them to boss in the same formation

**Comment by Jildor** [comment]
Apart from some boss script, would there be NPCs with smartAIs that use the CallForHelp action with ignoreLoS param?
If positive I can open a PR with the change, otherwise it might not be worth it?

**Comment by offl** [comment]
> would there be NPCs with smartAIs that use the CallForHelp action with ignoreLoS param

I doubt

If we want to solve this particular case right now without formations, the best option is CallForHelp with ignoreLoS, yes
But it's indeed will be used in few cases

> 4. even if 3 will be fixed right now I don't know how to use formations to link packs of trash and at the same time link them to boss in the same formation

https://github.com/TrinityCore/TrinityCore/issues/24764 but again can't be used always

We are far from correct implementation for things related to linked aggro in general and especially for linked aggro in case of bosses

**Pull request closed without merging**